java/maven/src/org/netbeans/modules/maven/options/MavenSettings.java

```diff
@@ -480,6 +484,19 @@ public void setBinaryDownloadStrategy(DownloadStrategy ds) {
             }
         }
 
+        Pattern mavenCoreJarPattern = Pattern.compile("maven-core-(\\d+\\.\\d+\\.\\d+)\\.jar");  // NOI18N
+        try (Stream<String> mavenCoreVersions = Files.list(lib)
+                .map(file -> file.getFileName())
```

Comment:
could map to `file.getFileName().toString()` since the next two steps work with Strings (remove `toString()` there).

javadoc of `getFileName` says "a path representing the name of the file or directory, or null if this path has zero elements", so we hopefully don't have to worry about null here (famous last words before NPE :)).

Reply:
Good spot!

Can you imagine a scenario where the path has zero elements? Sounds like quite odd to me.